pyblish_lite/control.py

```diff
@@ -58,40 +66,107 @@ def __init__(self, parent=None):
+        self.context.data["_type"] = 'context'
 
-        self.pair_generator = None
-        self.current_pair = (None, None)
+        self.context.data["icon"] = 'book'
+
+        self.context.data["_has_succeeded"] = False
+        self.context.data["_has_failed"] = False
+        self.context.data["_is_idle"] = True
+
+        self.context.data["__families__"] = ('__context__',)
+
+    def reset(self):
+        """
+        Discover plug-ins and run collection
+        - prepare_for_reset should be run before
+            - was split because of `Context` in instance list
+            - reset logic should be used only in window.py in `reset` method
+        """
+        # This is just backup
+        if self.context:
+            # - Context probably won't be shown in instance list
+            self.prepare_for_reset()
+
+        # Load collectors
+        self.load_plugins()
         self.current_error = None
+        # Process collectors load rest of plugins with collected instances
+        self.collect()
 
         self.processing = {
             "nextOrder": None,
             "ordersWithError": set()
         }
 
-        self._load()
-        self._run(until=pyblish.api.CollectorOrder,
-                  on_finished=self.was_reset.emit)
+    def load_plugins(self):
+        self.test = pyblish.logic.registered_test()
+        self.processing = {
+            "nextOrder": None,
+            "ordersWithError": set()
+        }
 
-    def validate(self):
-        self._run(until=pyblish.api.ValidatorOrder,
-                  on_finished=self.on_validated)
+        collectors = []
+        validators = []
+        extractors = []
+        conforms = []
+        plugins = pyblish.api.discover()
+
+        targets = pyblish.logic.registered_targets() or ["default"]
+        plugins = pyblish.logic.plugins_by_targets(plugins, targets)
+
+        for plugin in plugins:
+            if plugin.order < (pyblish.api.CollectorOrder + 0.5):
+                collectors.append(plugin)
+            elif plugin.order < (pyblish.api.ValidatorOrder + 0.5):
+                validators.append(plugin)
+            elif plugin.order < (pyblish.api.ExtractorOrder + 0.5):
+                extractors.append(plugin)
+            else:
+                conforms.append(plugin)
+
+        self.plugins = {
+            self.PART_COLLECT: collectors,
+            self.PART_VALIDATE: validators,
+            self.PART_EXTRACT: extractors,
+            self.PART_CONFORM: conforms
+        }
 
-    def publish(self):
-        self._run(on_finished=self.on_published)
+        self.was_discovered.emit()
+
+    def on_collected(self):
+        self.current_error = None
+        self.was_reset.emit()
+        self.was_finished.emit()
 
     def on_validated(self):
         pyblish.api.emit("validated", context=self.context)
         self.was_validated.emit()
+        self.validated = True
+        self.was_finished.emit()
+        if self.publishing:
```

Comment:
Is `self.publishing` ever False? If not, can we remove it?

Reply:
Is set to False on reset on init. Must stay